### PR TITLE
[improvement] support resource_group_location in runbook

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -465,14 +467,20 @@ def _deploy_environment(self, environment: Environment, log: Logger) -> None:
         else:
             try:
                 if self._azure_runbook.deploy:
+                    if self._azure_runbook.resource_group_location:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: The env resource group can be the same as first node.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 2]
```diff
@@ -246,6 +246,8 @@ class AzurePlatformSchema:
 
     shared_resource_group_name: str = AZURE_SHARED_RG_NAME
     resource_group_name: str = field(default="")
+    # specify resource group location
+    resource_group_location: Optional[str] = field(default=None)
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: add one for shared_resource_group_location, and set its default value as `shared_resource_group_name`

File: lisa/sut_orchestrator/azure/platform_.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: fixed

File: lisa/sut_orchestrator/azure/platform_.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: fixed

File: lisa/sut_orchestrator/azure/platform_.py [hunk 3]
```diff
@@ -464,6 +466,10 @@ def _deploy_environment(self, environment: Environment, log: Logger) -> None:
             log.info(f"dry_run: {self._azure_runbook.dry_run}")
         else:
             try:
+                location, deployment_parameters = self._create_deployment_parameters(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:469]: Check if it affects when "deploy:false".

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py:469]: I didn't see it affects when use existing rg.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**